eng/pipelines/coreclr/templates/sign-diagnostic-files.yml

```diff
@@ -4,7 +4,7 @@ parameters:
   timeoutInMinutes: ''
 
 steps:
-- ${{ if and(eq(parameters.isOfficialBuild, true), ne(variables['Build.Reason'], 'PullRequest')) }}:
+- ${{ if and(eq(parameters.isOfficialBuild, true), ne(variables['Build.Reason'], 'PullRequest'), or(startswith(variables['Build.SourceBranch'], 'refs/heads/release/'), startswith(variables['Build.SourceBranch'], 'refs/heads/internal/release/'))) }}:
```

Comment:
Is this removing signing from official builds out of main?  How's that related to this PR?

Reply:
It's not directly related, but it unblocks the ability to run internal builds on dev branches. I made a similar change in release/7.0 already. The issue is that the service connection is only available to approved branches. So running a dev build fails outright right now. The yaml should not attempt to use the connection.

Reply:
cc @hoyosjs

Reply:
https://github.com/dotnet/runtime/pull/74942

Reply:
So this is what caused that failure! I ended up doing a PR into the internal mirror to work around this. Glad to know it wasn’t intentional.